Feature/setup skeleton database

shadid-reza opened the pull request:
- Implemented n-tier architecture 
- Setup EF core
- Migrations
- Added dependencies and project reference

Initial database schema for the app.  
- Added core entities: Patient, Doctor, Hospital, Receptionist, and Appointment.  
- Introduced DoctorHospital as a junction table for doctor-hospital relationships.  
- Optimized visiting times with DoctorVisitingTime, allowing multiple time slots per doctor at a hospital while grouping shared schedules.  
- Established necessary one-to-many and many-to-many relationships.

Commit:
created a new structure

Commit:
implemented n-tier architecture

Warhammer4000 commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/warhammer4000/pr-checks/4b184c05-9790-463d-bd68-d265562a29c5?refs=4dff993b2260d2326253b9eba9d32db208cca185&source=prChecksComment)

Commit:
Initial database schema for the app.
- Added core entities: Patient, Doctor, Hospital, Receptionist, and Appointment.
- Introduced DoctorHospital as a junction table for doctor-hospital relationships.
- Optimized visiting times with DoctorVisitingTime, allowing multiple time slots per doctor at a hospital while grouping shared schedules.
- Established necessary one-to-many and many-to-many relationships.

Commit:
fixed error

shadid-reza commented on API/ShasthoBondhu/ShasthoBondhu/appsettings.Development.json:
```diff
@@ -0,0 +1,12 @@
+    }
+  },
+
+  "ConnectionStrings": {
+    "ShasthoBondhuDbContext": "data source=.;Database=sqldb-shasthobondhu-dev;Trusted_Connection=True;MultipleActiveResultSets=False;App=EntityFramework;TrustServerCertificate=true;"
+  }
```
show me a preview

Commit:
used composite keys

Commit:
fixed nominee relationship

shadid-reza commented on API/ShasthoBondhu/ShasthoBondhu.Data/Entities/DoctorHospital.cs:
```diff
@@ -0,0 +1,25 @@
+{
+    public class DoctorHospital
+    {
+        public int Id { get; set; }
```
which is better you think? guid or int?
